### ISSUE TITLE
Extract text node but not sub-elements?

### ISSUE THREAD

**Issue by edgarogh** [ikenox/h2s-rs#81]
When parsing trees of `<ul>`/`<li>`, the label of the `<li></li>` is often a simple text node that precedes the inner `<ul>` of the tree element. Selecting text using `#[h2s]` also gives me the contents of the sub-`<ul>`s. How can I just select the text node?

### Example

```html
<ul>
  <li>
    Node 1
    <ul>
      <li>Child 1</li>
      <li>Child 2</li>
    </ul>
  </li>
  <li>
    Node 2
    <ul>
      <li>Child 3</li>
      <li>Child 4</li>
    </ul>
  </li>
</ul>
```

I would like to define:

```rust
struct TreeNode {
  #[h2s] // What do I write here?
  name: String,

  #[h2s(select = "ul > li")]
  children: Vec<TreeNode>,
}
```

I didn't find any documentation on the full syntax of the `#[h2s]` attribute.

**Comment by ikenox** [comment]
@edgarogh 
Thank you for asking.
I have released h2s v0.18 which contains a new feature for this case.
Now `#[h2s(extractor=ExtractNthChild(0))]` will solve your problem.
Please see a usage example: https://github.com/ikenox/h2s-rs/blob/main/examples/explicit_extractor.rs

Note: Some deep-seated problems still remain (e.g. cannot use `ExtractNthChild` for `Optional<String>` field).
Sometimes I may do some improvements to solve it.

**Comment by edgarogh** [comment]
That's really cool, thank you!
